code/parse/sexp.cpp

```diff
@@ -13064,6 +13065,36 @@ void sexp_hud_set_text(int n)
 	}
 }
 
+void sexp_hud_set_xstr(int n)
+{
+	bool is_nan, is_nan_forever;
+
+	auto gaugename = CTEXT(n);
+	auto text = CTEXT(CDR(n));
+	auto id = eval_num(CDR(n), is_nan, is_nan_forever);
```

Comment:
Just to check, I reckon perhaps `id` should use `CDDR(n)` instead of `CDR(n)` (otherwise it's the same as `text`)

Reply:
Correct. Fixed. Good catch.